pkg/apis/stork/v1alpha1/types.go

```diff
@@ -308,3 +315,253 @@ const (
 	// GroupSnapshotStageFinal is when all stages are done for the group snapshot
 	GroupSnapshotStageFinal GroupVolumeSnapshotStageType = "Final"
 )
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// StorageCluster represents a storage cluster
+type StorageCluster struct {
+	meta.TypeMeta   `json:",inline"`
+	meta.ObjectMeta `json:"metadata,omitempty"`
+	Spec            StorageClusterSpec   `json:"spec"`
+	Status          StorageClusterStatus `json:"status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// StorageClusterList is a list of StorageCluster
+type StorageClusterList struct {
+	meta.TypeMeta `json:",inline"`
+	meta.ListMeta `json:"metadata,omitempty"`
+	Items         []StorageCluster `json:"items"`
+}
+
+// StorageClusterSpec is the spec used to define a storage cluster
+type StorageClusterSpec struct {
+	// Placement configuration for the storage cluster nodes
+	Placement *PlacementSpec `json:"placement"`
+	// Image is docker image of the storage driver
+	Image string `json:"image"`
+	// Kvdb is the information of kvdb that storage driver uses
+	Kvdb *KvdbSpec `json:"kvdb"`
+	// CloudStorage details of storage in cloud environment.
+	// Keep this only at cluster level for now until support to change
+	// cloud storage at node level is added.
+	CloudStorage *CloudStorageSpec `json:"cloudStorage"`
+	// SecretsProvider is the name of secret provider that driver will connect to
+	SecretsProvider string `json:"secretsProvider"`
+	// CSIEndpoint is the csi endpoint for the driver
+	CSIEndpoint string `json:"csiEndpoint"`
+	// CallHome send cluster information for analytics
+	CallHome bool `json:"callHome"`
+	// CommonSpec that is present at both cluster and node level
+	CommonSpec
```

Comment:
Don't need to call everything a spec. Only things with status need a spec.
https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#spec-and-status